djstripe/models/base.py

```diff
@@ -380,6 +380,44 @@ def _create_from_stripe_object(
 
         return instance
 
+    @classmethod
+    def get_or_create(cls, **kwargs):
+        """ Get or create an object of this class."""
+
+        try:
+            return cls.objects.get(id=kwargs["id"]), False
+        except cls.DoesNotExist:
+            return cls.create(**kwargs), True
+
+    @classmethod
+    def create(cls, **kwargs):
+        api_kwargs = dict(kwargs)
+
+        amount_fields = [
+            "amount",
+            "amount_capturable",
+            "amount_due",
+            "amount_off",
+            "amount_paid",
+            "amount_received",
+            "amount_refunded",
+            "amount_remaining",
+            "amount_reversed",
+        ]
+
+        for field in amount_fields:
+            if field in api_kwargs:
+                api_kwargs[field] = int(api_kwargs[field] * 100)
```

Comment:
This should probably check the field type instead of the name:

eg get field from cls._meta and check for `isinstance(field, djstripe.fields.StripeDecimalCurrencyAmountField)`

Note that some fields called amount are djstripe.fields.StripeDecimalCurrencyAmountField, and therefore are already in cents.

Reply:
Thanks, I will update